lib/hooks/core/hook-http.js

```diff
@@ -51,13 +51,13 @@ function requestWrap(request) {
 
     var namespace = cls.getNamespace();
     var uri = uriFromOptions(options);
-    var currContext = cls.getRootContext().createChildSpanData(uri);
-    currContext.span.setLabel(TraceLabels.HTTP_METHOD_LABEL_KEY, options.method);
-    currContext.span.setLabel(TraceLabels.HTTP_URL_LABEL_KEY, uri);
+    var span = agent.startSpan(uri);
+    span.addLabel(TraceLabels.HTTP_METHOD_LABEL_KEY, options.method);
+    span.addLabel(TraceLabels.HTTP_URL_LABEL_KEY, uri);
     if (options.headers) {
       // Adding context to the headers lets us trace the request
       // as it makes it through other layers of the Google infrastructure
-      currContext.addContextToHeaders(options.headers);
+      span.spanData_.span.addContextToHeaders(options.headers);
```

Comment:
Yeah, this is a bit ugly. We don't have a mechanism to do this cleanly without reaching into the internals here. Can you open an issue for this?